src/platform/library/schedule/ll_schedule.c

```diff
@@ -28,140 +27,42 @@ DECLARE_SOF_UUID("ll-schedule", ll_sched_uuid, 0x77de2074, 0x828c, 0x4044,
 
 DECLARE_TR_CTX(ll_tr, SOF_UUID(ll_sched_uuid), LOG_LEVEL_INFO);
 
-struct ll_vcore {
-	struct list_item list; /* list of tasks in priority queue */
-	pthread_mutex_t list_mutex;
-	pthread_t thread_id;
-	int vcore_ready;
-	int core_id;
-};
-
-static int tick_period_us;
+/* list of all tasks */
+static struct list_item sched_list;
 
-/**
- * Implement an override of how cores defined in SOF topology
- * are mapped to host cores.
- *
- * Returns the host core number to which SOF DSP core0 should
- * be mapped to.
- */
-static unsigned int sof_host_core_base(void)
+void schedule_ll_run_tasks(void)
 {
-	const char *host_core_env = getenv("SOF_HOST_CORE0");
-	unsigned int host_core = 0;
-
-	if (host_core_env) {
-		host_core = atoi(host_core_env);
-		tr_dbg(&ll_tr, "Mapping topology core ids to host cores %d...coreN\n", host_core);
-	}
-
-	return host_core;
-}
-
-static void *ll_thread(void *data)
-{
-	struct ll_vcore *vc = data;
-	struct timespec ts, td0, td1;
 	struct list_item *tlist, *tlist_;
 	struct task *task;
-	int err;
-	uint64_t delta;
-	cpu_set_t cpuset;
-	pthread_t thread;
-
-	/* Set affinity mask to pipeline core */
-	thread = pthread_self();
-	CPU_ZERO(&cpuset);
-	CPU_SET(vc->core_id, &cpuset);
-
-	printf("ll_schedule: new thread for core %d\n", vc->core_id);
-	err = pthread_setaffinity_np(thread, sizeof(cpuset), &cpuset);
-	if (err != 0)
-		fprintf(stderr, "error: failed to set CPU affinity to core %d: %s\n",
-			vc->core_id, strerror(err));
-
-	/* convert uS to seconds and ns */
-	ts.tv_sec = tick_period_us / 1000000;
-	ts.tv_nsec = (tick_period_us % 1000000) * 1000;
 
-	while (1) {
-		/*
-		 * The LL scheduler works with a periodic tick which we emulate
-		 * here to provide a similar processing experience on testbench
-		 * to actual DSP FW.
-		 */
-		if (tick_period_us) {
-			while (1) {
-				/* wait for next tick */
-				err = nanosleep(&ts, &ts);
-				if (err == 0)
-					break; /* sleep fully completed */
-				else if (err == EINTR) {
-					continue; /* interrupted - keep going */
-				} else {
-					/* something bad happened ... */
-					fprintf(stderr, "error: sleep failed: %s\n",
-						strerror(err));
-					goto out;
-				}
-			}
-		}
-
-		/* LL time slice now running at this point */
-		pthread_mutex_lock(&vc->list_mutex);
-
-		/* list empty then return */
-		if (list_is_empty(&vc->list)) {
-			pthread_mutex_unlock(&vc->list_mutex);
-			fprintf(stdout, "LL scheduler thread exit - list empty\n");
-			break;
-		}
-
-		/* iterate through the task list */
-		list_for_item_safe(tlist, tlist_, &vc->list) {
-			task = container_of(tlist, struct task, list);
+	/* list empty then return */
+	if (list_is_empty(&sched_list))
+		fprintf(stdout, "LL scheduler thread exit - list empty\n");
```

Comment:
will this change merge? 
if merge, old logic is gone, does it means after that sof does not support non zephyr build anymore?

Reply:
This is for library simplified version only so no worry.

Reply:
ok， sorry, sof have many files with same name, I did not check it clearly.